dev-packages/application-package/src/application-props.ts

```diff
@@ -36,6 +36,50 @@ export namespace NpmRegistryProps {
     };
 }
 
+/**
+ * Base configuration for the Theia application.
+ */
+export interface ApplicationConfig {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    readonly [key: string]: any;
+}
+
+/**
+ * Application configuration for the frontend. The following properties will be injected into the `index.html`.
+ */
+export interface FrontendApplicationConfig extends ApplicationConfig {
+
+    /**
+     * The default theme for the application. If not given, defaults to `dark`. If invalid theme is given, also defaults to `dark`.
+     */
+    readonly defaultTheme?: string;
+
+    /**
+     * The default icon theme for the application. If not given, defaults to `none`. If invalid theme is given, also defaults to `none`.
+     */
+    readonly defaultIconTheme?: string;
+
+    /**
+     * The name of the application. `Eclipse Theia` by default.
+     */
+    readonly applicationName?: string;
+
+    /**
+     * Electron specific configuration.
+     */
+    readonly electron?: Readonly<ElectronFrontendApplicationConfig>;
+}
```

Comment:
I believe the following changes are unnecessary, we would only want to make the properties optional, not move the interfaces around in the same file.

Reply:
I moved the interface because it made more sense to have it defined before being used.

Fixed in the last commit.